runtime/drivers/athena/athena.go

```diff
@@ -0,0 +1,294 @@
+	return nil, false
+}
+
+// Repo implements drivers.Connection.
+func (c *Connection) AsRepoStore(instanceID string) (drivers.RepoStore, bool) {
+	return nil, false
+}
+
+// OLAP implements drivers.Connection.
+func (c *Connection) AsOLAP(instanceID string) (drivers.OLAPStore, bool) {
+	return nil, false
+}
+
+// Migrate implements drivers.Connection.
+func (c *Connection) Migrate(ctx context.Context) (err error) {
+	return nil
+}
+
+// MigrationStatus implements drivers.Connection.
+func (c *Connection) MigrationStatus(ctx context.Context) (current, desired int, err error) {
+	return 0, 0, nil
+}
+
+// AsObjectStore implements drivers.Connection.
+func (c *Connection) AsObjectStore() (drivers.ObjectStore, bool) {
+	return c, true
+}
+
+// AsTransporter implements drivers.Connection.
+func (c *Connection) AsTransporter(from, to drivers.Handle) (drivers.Transporter, bool) {
+	return nil, false
+}
+
+func (c *Connection) AsFileStore() (drivers.FileStore, bool) {
+	return nil, false
+}
+
+// AsSQLStore implements drivers.Connection.
+func (c *Connection) AsSQLStore() (drivers.SQLStore, bool) {
+	return nil, false
+}
+
+// DownloadFiles returns a file iterator over objects stored in gcs.
+// The credential json is read from config google_application_credentials.
+// Additionally in case `allow_host_credentials` is true it looks for "Application Default Credentials" as well
+func (c *Connection) DownloadFiles(ctx context.Context, source *drivers.BucketSource) (drivers.FileIterator, error) {
+	conf, err := parseSourceProperties(source.Properties)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse config: %w", err)
+	}
+
+	prefix := "parquet_output_" + uuid.New().String()
+	bucketName := strings.TrimPrefix(strings.TrimRight(conf.OutputLocation, "/"), "s3://")
+	unloadPath := bucketName + "/" + prefix
+	err = c.unload(ctx, conf, "s3://"+unloadPath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to unload: %w", err)
+	}
+
+	bucketObj, err := c.openBucket(ctx, conf, bucketName)
+	if err != nil {
+		return nil, fmt.Errorf("cannot open bucket %q: %w", unloadPath, err)
+	}
+
+	opts := rillblob.Options{
+		ExtractPolicy: &runtimev1.Source_ExtractPolicy{
+			// FilesStrategy: runtimev1.Source_ExtractPolicy_STRATEGY_HEAD,
+		},
+		GlobPattern: prefix + "/**",
+	}
+
+	it, err := rillblob.NewIterator(ctx, bucketObj, opts, c.logger)
+	if err != nil {
+		// TODO :: fix this for single file access. for single file first call only happens during download
+		var failureErr awserr.RequestFailure
+		if !errors.As(err, &failureErr) {
+			return nil, fmt.Errorf("failed to create the iterator %q %w", unloadPath, err)
+		}
+
+		// check again
+		if errors.As(err, &failureErr) && (failureErr.StatusCode() == http.StatusForbidden || failureErr.StatusCode() == http.StatusBadRequest) {
+			return nil, drivers.NewPermissionDeniedError(fmt.Sprintf("can't access remote err: %v", failureErr))
+		}
+	}
```

Comment:
Are these conditions needed here? IIRC, they are for handling public bucket access, which is not needed for Athena reads. I would suggest checking with @k-anshul.